assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js

```diff
@@ -0,0 +1,177 @@
+        <ContentWrapper>{option.label}</ContentWrapper>
+        {option.animation?.Effect ? (
+          <option.animation.Effect>{option.label}</option.animation.Effect>
+        ) : (
+          <NoEffect>{option.label}</NoEffect>
+        )}
+      </Tooltip>
+    </AnimationListItem>
+  );
+});
+export default function EffectChooserDropdown({
+  onAnimationSelected,
+  onNoEffectSelected,
+  isBackgroundEffects = false,
+  selectedEffectType,
+  disabledTypeOptionsMap,
+  direction,
+}) {
+  const { enableExperimentalAnimationEffects } = useFeatures();
+
+  const selectedValue = useMemo(
+    () => getDirectionalEffect(selectedEffectType, direction),
+    [selectedEffectType, direction]
+  );
+  // BACKGROUND EFFECT SET UP
+  const getDisabledBackgroundEffects = useCallback(() => {
+    const disabledDirectionalEffects = Object.entries(disabledTypeOptionsMap)
+      .map(([effect, val]) => [effect, val.options])
+      .reduce(
+        (directionalEffects, [effect, directions]) => [
+          ...directionalEffects,
+          ...(directions || []).map((dir) => getDirectionalEffect(effect, dir)),
+        ],
+        []
+      );
+    return Object.keys(backgroundEffectOptions).filter((directionalEffect) =>
+      disabledDirectionalEffects.includes(directionalEffect)
+    );
+  }, [disabledTypeOptionsMap]);
+
+  const disabledBackgroundEffects = getDisabledBackgroundEffects();
+
+  // END BACKGROUND EFFECT SET UP
+  const animationOptionsObject = isBackgroundEffects
+    ? backgroundEffectOptions
+    : foregroundEffectOptions;
+
+  // remove experiments if needed
+  const availableAnimationOptions = useMemo(() => {
+    return enableExperimentalAnimationEffects
+      ? Object.values(animationOptionsObject)
+      : Object.values(animationOptionsObject).filter(({ value }) => {
+          return experimentalEffects.indexOf(value) === -1;
+        });
+  }, [animationOptionsObject, enableExperimentalAnimationEffects]);
+
+  console.log(availableAnimationOptions);
+  const assembledOptions = availableAnimationOptions?.map((option) => {
+    const isDisabled =
+      isBackgroundEffects && disabledBackgroundEffects.includes(option.value);
+    return {
+      ...option,
+      disabled: isDisabled,
+      tooltip:
+        isDisabled && disabledTypeOptionsMap[option.animation?.value]?.tooltip,
+    };
+  });
+
+  const handleSelect = useCallback((event, value) => {
+    event.preventDefault();
+    if (value === NO_ANIMATION) {
+      onNoEffectSelected();
+    }
+
+    const selectedAnimation = animationOptionsObject[value].animation;
+    onAnimationSelected({
+      animation: selectedAnimation.value,
+      panDir: selectedAnimation.panDirection,
+      zoomDirection: selectedAnimation.zoomDirection,
+      flyInDir: selectedAnimation.flyInDirection,
+      rotateInDir: selectedAnimation.rotateInDirection,
+      whooshInDir: selectedAnimation.whooshInDirection,
+      scaleDirection: selectedAnimation.scaleDirection,
+    });
```

Comment:
nice. didn't think to just pass everything, but totally works and simplifies things 🎉

Reply:
I played with the idea of checking for these keys and then passing what was found but kind of liked how readable this is to track animation options?

Reply:
Yea completely agree, just lets them all be handled the same. And their creator functions discern arguments they're intaking anyway so no worries there. v nice